test/cpp/end2end/orca_service_end2end_test.cc

```diff
@@ -107,16 +112,13 @@ class OrcaServiceEnd2endTest : public ::testing::Test {
   ~OrcaServiceEnd2endTest() override { server_->Shutdown(); }
 
   std::string server_address_;
-  OrcaService orca_service_;
+  ServerMetricRecorder server_metric_recorder_;
+  std::unique_ptr<OrcaService> orca_service_;
```

Comment:
I don't think this needs to be changed to be a `unique_ptr<>`.  I think it's fine being a direct data member.

Reply:
I did this since we now require ServerMetricRecorder and wanted to avoid having an implicitly default-created instance. I deleted the default c'tor kept it for now.

Reply:
There is no need to use dynamic allocation to pass the parameter here.  You can set the parameter via the ctor's initialization list:

```
class MyClass {
 public:
  MyClass() : orca_service_(&server_metric_recorder_) {}

 private:
  ServerMetricRecorder server_metric_recorder_;
  OrcaService orca_service_;
};
```

Reply:
Done.

Reply:
Done.